flight/Modules/VtolPathFollower/vtol_follower_control.c

```diff
@@ -58,6 +58,69 @@ static VtolPathFollowerSettingsData guidanceSettings;
 static AltitudeHoldSettingsData altitudeHoldSettings;
 struct pid vtol_pids[VTOL_PID_NUM];
 
+static inline float vtol_interpolate(const float fraction, const float beginVal,
+	const float endVal) {
+	return beginVal + (endVal - beginVal) * bound_min_max(fraction, 0, 1);
+}
+
+static inline float vtol_magnitude(const float *v, int n) {
+	float sum=0;
+
+	for (int i=0; i<n; i++) {
+		sum += powf(v[i], 2);
+	}
+
+	return sqrtf(sum);
+}
+
+static inline void vtol_calculate_distances(const float *actual,
+	const float *desired, float *out, int normalize) {
+	out[0] = desired[0] - actual[0];
+	out[1] = desired[1] - actual[1];
+	out[2] = desired[2] - actual[2];
+
+	if (normalize) {
+		float mag=vtol_magnitude(out, 3);
+
+		out[0] /= mag;  out[1] /= mag;  out[2] /= mag;
+	}
+}
+
+static inline void vtol_limit_velocity(float *vels, float limit) {
+	float mag=vtol_magnitude(vels, 2);	// only horiz component
+	float scale = mag / limit;
+
+	if (scale > 1) {
+		vels[0] /= mag;
+		vels[1] /= mag;
+	}
+}
+
+/* "Real" deadbands are evil.  Control systems end up fighting the edge.
+ * You don't teach your integrator about emerging drift.  Discontinuities
+ * in your control inputs cause all kinds of neat stuff. */
+static inline float vtol_deadband(float in, float w, float b) {
+	/* So basically.. we want the function to be tangent to the
+	** linear sections-- have a slope of 1-- at -w and w.  In the
+	** middle we want a slope of b.   So the cube here does all the
+	** work b isn't doing. */
+	float m = cbrtf((1-b)/(3*powf(w,2)));
```

Comment:
following discussion on IRC b and w should probably come from some "effective deadband" setting in VtolPathFollowerSettings that is specified in meters and then can be precomputed/cached when settings change.